### PR TITLE
Add docs to show how to use custom images 

### PR DESCRIPTION
**Pull request by mrodm**
This PR includes documentation to set overrides for some docker images. Currently, the docker images that can be overwritten are:
- Elasticsearch
- Kibana
- Elastic Agent

This overwrites can be achieved in two different ways:
- Using Environment Variables
- Using the configuration file (`~/.elastic-package/config.yml`)

Link to the new doc has been added to the description of `elastic-package stack up`.

## How to test locally

As an example, it is going to be set an overwrite for Elastic Agent ([docs link](https://github.com/elastic/elastic-agent#packaging)):

```bash
# Go to the elastic-agent repository
cd /path/to/elastic-agent
git checkout 8.8
DEV=true SNAPSHOT=true EXTERNAL=true PLATFORMS=linux/amd64 PACKAGES=docker  mage -v package
docker tag docker.elastic.co/beats/elastic-agent-complete:8.8.2-SNAPSHOT docker.elastic.co/beats/elastic-agent-complete:8.8.2-SNAPSHOT-test

git checkout v8.8.1
DEV=true SNAPSHOT=true EXTERNAL=true PLATFORMS=linux/amd64 PACKAGES=docker  mage -v package
docker tag docker.elastic.co/beats/elastic-agent-complete:8.8.1-SNAPSHOT docker.elastic.co/beats/elastic-agent-complete:8.8.1-SNAPSHOT-test
```

As this image is used by both Elastic Agent and Fleet Server services, `docker ps` will show two containers running the custom image.

- Using environment variables:
  ```bash
  export ELASTIC_AGENT_IMAGE_REF_OVERRIDE=docker.elastic.co/beats/elastic-agent-complete:8.8.2-SNAPSHOT-test
  elastic-package stack up -v --version 8.8.2-SNAPSHOT
  docker ps --format "{{.Image}}"
  ```

- Using configuration file
    - Configuration file should be like:
      ```yaml
      stack:
          image_ref_overrides:
            8.8.2-SNAPSHOT:
              elastic-agent: "docker.elastic.co/beats/elastic-agent-complete:8.8.2-SNAPSHOT-test"
      profile:
          current: default
      ```
    - steps to follow
      ```bash
      unset ELASTIC_AGENT_IMAGE_REF_OVERRIDE
      elastic-package stack up -v -d --version 8.8.2-SNAPSHOT
      docker ps --format "{{.Image}}"
      ```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -360,7 +360,7 @@ Boot up the stack.
 
 _Context: global_
 
-Use this command to spin up a Docker-based Elastic Stack consisting of Elasticsearch, Kibana, and the Package Registry. By default the latest released version of the stack is spun up but it is possible to specify a different version, including SNAPSHOT versions by appending --version <version>.
+Use this command to spin up a Docker-based Elastic Stack consisting of Elasticsearch, Kibana, and the Package Registry. By default the latest released version of the stack is spun up but it is possible to specify a different version, including SNAPSHOT versions by appending --version <version>. In case, you need to run your own custom images (e.g. Elasticsearch, Kibana or elastic-agent), see [this document](./docs/howto/custom_images.md).
```
Comment (jsoriano) [inline README.md:363]: Nit.
```suggestion
Use this command to spin up a Docker-based Elastic Stack consisting of Elasticsearch, Kibana, and the Package Registry. By default the latest released version of the stack is spun up but it is possible to specify a different version, including SNAPSHOT versions by appending --version <version>.

You can run your own custom images for Elasticsearch, Kibana or elastic-agent, see [this document](./docs/howto/custom_images.md) for more information.
```

File: README.md [hunk 2]
```diff
@@ -396,11 +396,11 @@ _Context: global_
 
 Use this command to boot up the stack locally.
 
-By default the latest released version of the stack is spun up but it is possible to specify a different version, including SNAPSHOT versions by appending --version <version>.
+By default the latest released version of the stack is spun up but it is possible to specify a different version, including SNAPSHOT versions by appending --version <version>. In case, you need to run your own custom images (e.g. Elasticsearch, Kibana or elastic-agent), see [this document](./docs/howto/custom_images.md).
```
Comment (jsoriano) [inline README.md:399]: Nit.
```suggestion
By default the latest released version of the stack is spun up but it is possible to specify a different version, including SNAPSHOT versions by appending --version <version>.

You can run your own custom images for Elasticsearch, Kibana or elastic-agent, see [this document](./docs/howto/custom_images.md) for more information.
```

File: cmd/stack.go [hunk 3]
```diff
@@ -26,19 +26,19 @@ var availableServices = map[string]struct{}{
...
+By default the latest released version of the stack is spun up but it is possible to specify a different version, including SNAPSHOT versions by appending --version <version>. In case, you need to run your own custom images (e.g. Elasticsearch, Kibana or elastic-agent), see [this document](./docs/howto/custom_images.md).
 
 Be aware that a common issue while trying to boot up the stack is that your Docker environments settings are too low in terms of memory threshold.
 
-To expose local packages in the Package Registry, build them first and boot up the stack from inside of the Git repository containing the package (e.g. elastic/integrations). They will be copied to the development stack (~/.elastic-package/stack/development) and used to build a custom Docker image of the Package Registry.
+For Elastic stack versions prior to 8.7, to expose local packages in the Package Registry, build them first and boot up the stack from inside of the Git repository containing the package (e.g. elastic/integrations). They will be copied to the development stack (~/.elastic-package/stack/development) and used to build a custom Docker image of the Package Registry.
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply cmd/stack.go:41]: From 8.7 the same instructions apply to expose local packages in the Package Registry, isn't it?

Would you like to mean here that from 8.7 this is not needed to test packages?

File: docs/howto/custom_images.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+Setting `--version` flag, you can start the Elastic stack using different versions, for instance SNAPSHOT version (e.g. 8.8.2-SNAPSHOT).
+If this flag is not set, the default version used is by elastic-package is defined [here](../../internal/install/stack_version.go).
+
+There could be cases where you need to use your own custom images for development or debugging purposes.
+If you need to use your own custom docker images for your service (e.g. elastic-agent), this could be
+achieved in two different ways:
+- [defining environment variables](#using-environment-variables)
+- [updating elastic-package configuration file](#using-the-configuration-file)
+
+If both way are used, environment variables have preference to set the custom image.
```
Comment (jsoriano) [inline docs/howto/custom_images.md:14]:
```suggestion
If both ways are used, environment variables have preference to set the custom image.
```

**Review by jsoriano — COMMENTED** [review]
Nice, it is great to have documentation about this feature.

File: cmd/stack.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrodm) [reply cmd/stack.go:41]: Right, same instructions apply to expose local packages in the Elastic Package Registry.

I've rephrased this paragraph to indicate that this is not needed to run the tests, starting with Elastic stack >= 8.7.0

File: README.md [hunk 5]
```diff
@@ -362,6 +362,8 @@ _Context: global_
 
 Use this command to spin up a Docker-based Elastic Stack consisting of Elasticsearch, Kibana, and the Package Registry. By default the latest released version of the stack is spun up but it is possible to specify a different version, including SNAPSHOT versions by appending --version <version>.
 
+You can run your own custom images for Elasticsearch, Kibana or elastic-agent, see [this document](./docs/howto/custom_images.md).
```
Comment (jsoriano) [inline README.md:365]: Nit. Actually if we use the uppercased names of the products we should probably use Elastic Agent.
```suggestion
You can run your own custom images for Elasticsearch, Kibana or Elastic Agent, see [this document](./docs/howto/custom_images.md).
```

File: README.md [hunk 6]
```diff
@@ -398,9 +400,11 @@ Use this command to boot up the stack locally.
 
 By default the latest released version of the stack is spun up but it is possible to specify a different version, including SNAPSHOT versions by appending --version <version>.
 
+You can run your own custom images for Elasticsearch, Kibana or elastic-agent, see [this document](./docs/howto/custom_images.md).
```
Comment (jsoriano) [inline README.md:403]: Nit.
```suggestion
You can run your own custom images for Elasticsearch, Kibana or Elastic Agent, see [this document](./docs/howto/custom_images.md).
```

**Review by jsoriano — APPROVED** [review]
:rocket:

**Pull request merged**